include/nbl/builtin/hlsl/cpp_compat/impl/intrinsics_impl.hlsl

```diff
@@ -276,7 +313,10 @@ struct lerp_helper<T, bool>
 {
 	static inline T lerp(NBL_CONST_REF_ARG(T) x, NBL_CONST_REF_ARG(T) y, NBL_CONST_REF_ARG(bool) a)
 	{
-		return a ? y : x;
+		if (a)
+			return y;
+		else
+			return x;
 	}
 };
```

Comment:
@Fletterio you can add a partial spec that uses `?` for `is_fundamental_v` if it means that much to you

Reply:
If there's no perf hit I'll just use this one